mlrun/runtimes/pod.py

```diff
@@ -436,52 +442,69 @@ def _clear_affinity_if_initialized_but_empty(func):
         @functools.wraps(func)
         def wrapper(self, *args, **kwargs):
             resp = func(self, *args, **kwargs)
-            if not self.affinity:
-                self.affinity = None
+            affinity_field_name = kwargs.get("affinity_field_name", "affinity")
+            self_affinity = getattr(self, affinity_field_name)
+            if not getattr(self, affinity_field_name):
+                setattr(self, affinity_field_name, None)
```

Comment:
A matter of style, but my feeling is that this wrapper is becoming too much hacky and magic, I would replace it with regular function calls